### ISSUE TITLE
Provide Trained Embedding Model and Umap and Only Train After

### ISSUE THREAD

**Issue by scharfm16** [MaartenGr/BERTopic#1182]
Hi @MaartenGr,

First of all thank you so much for this amazing package!

Quick Question: is there a way to provide already fit embedding and umap and then only fit kmeans and topic label creation after?

**Comment by MaartenGr** [comment]
Yes, you will have to do something like the following:

```python
from sklearn.datasets import fetch_20newsgroups
from sentence_transformers import SentenceTransformer
from umap import UMAP
from bertopic import BERTopic
from bertopic.dimensionality import BaseDimensionalityReduction

# Prepare embeddings and reduce them
docs = fetch_20newsgroups(subset='all',  remove=('headers', 'footers', 'quotes'))['data']
sentence_model = SentenceTransformer("all-MiniLM-L6-v2")
embeddings = sentence_model.encode(docs, show_progress_bar=False)
reduced_embeddings = UMAP(n_neighbors=15, n_components=5, min_dist=0.0, metric='cosine').fit_transform(embeddings)

# Train our topic model using our pre-trained sentence-transformers embeddings
empty_dimensionality_model = BaseDimensionalityReduction()
topic_model = BERTopic(umap_model=empty_dimensionality_model )
topics, probs = topic_model.fit_transform(docs, reduced_embeddings)
```

You can find more about that in the documentation [here](https://maartengr.github.io/BERTopic/getting_started/dim_reduction/dim_reduction.html) and [here](https://maartengr.github.io/BERTopic/getting_started/embeddings/embeddings.html#custom-embeddings).
